modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/TableScanNode.java

```diff
@@ -184,7 +184,13 @@ private void push() throws Exception {
             }
         }
 
+        assert waiting >= 0;
+
         if (waiting == 0 || activeSubscription == null) {
+            if (activeSubscription == null) {
+                waiting = 0;
```

Comment:
why do you need to reset the counter here?

Reply:
ok i replace it into more obvious place, plz check once more.